src/routes/movies.py

```
@@ -1,13 +1,181 @@
+            if not language:
+                language = LanguageModel(name=language_name)
+                db.add(language)
+                await db.flush()
+            languages.append(language)
+        new_movie = MovieModel(
+            name=movie_data.name,
+            date=movie_data.date,
+            score=movie_data.score,
+            overview=movie_data.overview,
+            status=movie_data.status,
+            budget=movie_data.budget,
+            revenue=movie_data.revenue,
+            country=country,
+            genres=genres,
+            actors=actors,
+            languages=languages,
+        )
+        db.add(new_movie)
+        await db.commit()
+
+        return MovieDetailSchema.model_validate(new_movie)
+    except IntegrityError:
+        await db.rollback()
+
+        raise HTTPException(status_code=400, detail="Invalid input data.")
+
+
+@router.delete("/movies/{movie_id}/", status_code=204)
+async def delete_movie(movie_id: int, db: AsyncSession = Depends(get_db)):
+    result = await db.scalar(select(MovieModel).where(MovieModel.id == movie_id))
+    if not result:
+        raise HTTPException(status_code=404, detail="Movie with the given ID was not found.")
+    await db.delete(result)
+    await db.commit()
+    return None
+
+
+@router.patch("/movies/{movie_id}/", status_code=200, )
+async def update_movie(movie_id: int, movie: MovieUpdateSchema, db: AsyncSession = Depends(get_db)):
+    result = await db.scalar(select(MovieModel).where(MovieModel.id == movie_id))
+    if not result:
+        raise HTTPException(status_code=404, detail="Movie with the given ID was not found.")
+    try:
+        update_data = movie.model_dump(exclude_unset=True)
+    except ValueError:
+        raise HTTPException(status_code=400, detail="Invalid input data.")
+    for key, value in update_data.items():
+        setattr(result, key, value)
```

After gathering `update_data` you set attributes on the ORM object blindly (`setattr(result, key, value)`). Even if schema currently omits relationship fields, be explicit: only allow the permitted scalar fields to be updated (name, date, score, overview, status, budget, revenue). This prevents accidental updates if schema changes in the future. Also perform validations (see previous comment) before applying changes so you can return the exact error message required by the spec if input is invalid.